lua/toggleterm/terminal.lua

```diff
@@ -325,13 +327,31 @@ end
 ---@private
 ---Pass self as first parameter to callback
 function Terminal:__stdout()
-  if self.on_stdout then return function(...) self.on_stdout(self, ...) end end
+  if self.auto_scroll or self.on_stdout then
+    return function(...)
+      if self.auto_scroll and vim.fn.bufexists(self.bufnr) == 1 then
```

Comment:
```suggestion
      if self.auto_scroll and api.nvim_buf_is_valid(self.bufnr) then
````

Reply:
Unless the specific check was providing something that the api function doesn't cover

Reply:
Great! Thanks! I did look for a native lua api for what I wanted and only found the `vim.fn` version